chatkit-core/src/main/kotlin/com/pusher/chatkit/rooms/RoomStore.kt

```diff
@@ -80,7 +80,10 @@ internal class RoomStore(
                 is UserSubscriptionEvent.AddedToRoomEvent ->
                     listOf(event.also { this += event.room })
                 is UserSubscriptionEvent.RoomUpdatedEvent ->
-                    listOf(event.also { this += event.room })
+                    listOf(event.also {
+                        //member user ids seem to get lost here, so append whatever we had saved last time
```

Comment:
We should be definite and precise in our comments - this comment suggests that we don't really understand what is happening and so adds uncertainty in the mind of the reader, rather than helping them understand.

Reply:
Additionally, this situation applies in several other parts of this method, where we take Room entities deserialised from the subscription and add (possibly replacing an existing one) to `this`. We should cover these cases as well, preferable with tests.

Tests for the initial state cases are more difficult to trigger, but I can help with approaching that - I think else where I've written unit tests to cover these cases rather than integration ones, because it's difficult to trigger a reconnection on demand in the integration tests.

Reply:
Updated the comment. If it helps, I'm not sure what's happening so it'd be good to understand maybe then why the server is not sending us back the user ids in this instance, so we can have a better understanding of how this fits together.

Happy to add some more tests but can you help me understand what we're aiming for those? We've already got a lot of tests that hit those cases so it might be worth checking what's there to start with.